db/changes.go

```diff
@@ -1308,9 +1359,16 @@ func createChangesEntry(ctx context.Context, docid string, db *DatabaseCollectio
 		return nil
 	}
 
-	changes := make([]ChangeRev, 1)
-	changes[0] = ChangeRev{"rev": populatedDoc.CurrentRev}
-	row.Changes = changes
+	row.Changes = []ChangeByVersionType{{ChangesVersionTypeRevTreeID: populatedDoc.CurrentRev}}
+	switch options.VersionType {
+	case ChangesVersionTypeCV:
+		row.Changes[0] = ChangeByVersionType{options.VersionType: populatedDoc.HLV.GetCurrentVersionString()}
+	case ChangesVersionTypeRevTreeID:
+		fallthrough
+	default:
+		// already initialized with a 'rev' change entry above
+	}
```

Comment:
slight preference for this code

```suggestion

	switch options.VersionType {
	case ChangesVersionTypeCV:
		row.Changes = []ChangeByVersionType{{options.VersionType: populatedDoc.HLV.GetCurrentVersionString()}}
	case ChangesVersionTypeRevTreeID:
	    row.Changes = []ChangeByVersionType{{ChangesVersionTypeRevTreeID: populatedDoc.CurrentRev}}
	default:
	     base.AssertfCtx(ctx, "createChangeEntry called with an unsupported VersionType: %s", options.VersionType)
	}
```